src/qt/bitcoingui.cpp

```diff
@@ -786,8 +797,13 @@ void BitcoinGUI::setNumBlocks(int count, const QDateTime& blockDate, double nVer
             if (header) {
                 return;
             }
-            // Case: not Importing, not Reindexing and no network connection
-            progressBarLabel->setText(tr("No block source available..."));
+            if (clientModel->getNumConnections() == 0)
+            {
+                // Case: not yet connected to peers
+                progressBarLabel->setText(tr("Connecting to peers..."));
+            }
+            else
+                updateHeadersSyncProgressLabel();
```

Comment:
I believe this is dead code - getBlockSource will always return at least BLOCK_SOURCE_NETWORK if getNumConnections is > 0.